### PR TITLE
Preapps 555 Choose calendar to add event to

### PR DESCRIPTION
**Pull request by itsmeprashant**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -476,6 +497,16 @@ export class ZimbraBatchClient {
 			)
 		);
 
+	public singleRequest = (options: JsonRequestOptions) => {
+		singleJSONRequest({
+			...options,
+			sessionId: this.sessionId,
+			origin: this.origin,
+			name: options.name,
```
Comment (silentsakky) [inline src/batch-client/index.ts]: below two statements are not required

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -476,6 +497,16 @@ export class ZimbraBatchClient {
 			)
 		);
 
+	public singleRequest = (options: JsonRequestOptions) => {
+		singleJSONRequest({
```
Comment (silentsakky) [inline src/batch-client/index.ts]: by calling this function directly we are skipping code for notifications handling

**Comment by dajay0** [comment]
Looked through. I dont have any comments.

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -162,20 +174,30 @@ export class ZimbraBatchClient {
...
+		appointment: CalendarItemInput
+	) =>
 		this.jsonRequest({
 			name: 'CreateAppointment',
 			body: {
 				...denormalize(CalendarItemCreateModifyRequest)(appointment)
-			}
+			},
+			accountName: accountName,
+			namespace: Namespace.Mail
```
Comment (pl12133) [inline src/batch-client/index.ts]: It looks like the namespace will default to `Namespace.Mail`, so would it be OK to remove this extra option?

Comment (itsmeprashant) [reply src/batch-client/index.ts]: Done!

**Pull request merged**